src/puppetlabs/trapperkeeper/config.clj

```diff
@@ -59,6 +59,12 @@
     #{".yaml" ".yml"}
     (yaml/parse-string (slurp file))))
 
+(defn override-restart-file-from-cli-data
+  [config-data cli-data]
+  (if-let [cli-restart-file (:restart-file cli-data)]
+    (assoc-in config-data [:global :restart-file] cli-restart-file)
```

Comment:
Do you think it would be worth adding a warning log message here if it's defined in both places?

I don't know if the logging subsystem is even initialized by this point, so, may not be a realistic option.